src/main/scala/com/advancedtelematic/campaigner/Boot.scala

```diff
@@ -49,14 +52,14 @@ object Boot extends BootApp
 
   log.info(s"Starting $version on http://$host:$port")
 
-  val director = new DirectorHttpClient(directorUri)
-
   val deviceRegistry = new DeviceRegistryHttpClient(deviceRegistryUri)
+  val director = new DirectorHttpClient(directorUri)
+  val resolver = new ResolverHttpClient(userProfileUri)
```

Comment:
I started out with that approach, but then it came to the point where I would need to instantiate resolvers in `UpdateResource`, where we don't have `ActorSystem` or `Materializer`.

Reply:
you have both of those, but this is not really a matter of approach, because you are instantiating a resolver with a user profile uri, this will not work.

Reply:
Ok, sorry, now I see it.

So, you need to create a separate user profile client in Boot, where you need to also instantiate a new Resolver client, which doesn't accept any uri. Then when you actually call resolver client you can pass along an uri.